docs/architecture/adr-007-specialization-groups.md

```diff
@@ -0,0 +1,178 @@
+# ADR 007: Specialization Groups
+
+## Changelog
+
+- 2019 Jul 31 : Initial Conception
+
+## Context
+
+This idea was first conceived of in order to fulfill the use case of the
+creation of a decentralized Computer Emergency Response Team (dCERT), Whose
+members would be elected by the Cosmos Community and would fulfill the role of
+coordinating the Cosmos community under an emergency situations. This thinking
+can be further abstracted into the conception of "blockchain specialization
+groups". 
+
+The creation of these groups are the beginning of specialization capabilities
+within a wider blockchain community which could be used to enable a certain
+level of delegated responsibilities. Examples of specialization which could be
+beneficial to a blockchain community include: code auditing, emergency response,
+code development etc. This type of community organization paves the way for
+individual stakeholders to delegate votes by issue type, if in the future
+governance proposals include a field for issue type. 
+
+
+## Decision
+
+A specialization group can be broadly broken down into the following functions
+(herein containing examples): 
+ - Membership Admittance
+ - Membership Acceptance
+ - Membership Revocation
+   - (probably) Without Penalty 
+     - member steps down (self-Revocation)
+     - replaced by new member from governance
+   - (probably) With Penalty 
```

Comment:
How do you propose penalties be executed? What is the general idea and mechanism behind this? Should we allude to this at all in the ADR?

Reply:
These are just general ideas as to mechanisms which many specialization groups will probably want - this is mostly a high level overview. I'm doubtful that there is a way to easily abstract this functionality into the interface level, however if that's the case we can add that as an addendum to this ADR. Otherwise this functionality is intended to be built by the instance of the interfaces proposed.  (so I don't think we should allude to any more specifics than we've given, at least for now)